app/task_routes.py

```diff
@@ -0,0 +1,190 @@
+        abort(make_response(jsonify({"Message":response_str}), 400))
+    
+    matching_task = Task.query.get(task_id)
+
+    if not matching_task:
+        response_str = f"Task with id #{task_id} was not found in the database."
+        abort(make_response(jsonify({"Message": response_str}), 404))
+
+    return matching_task
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+    
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":        
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_param is None:
+        tasks = Task.query.all()
+    elif sort_param == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+
+    response = []
+    for task in tasks:
+        task_dict = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False
+            }
+        response.append(task_dict)
+    return jsonify(response), 200
+    
+    
+    name_param = request.args.get("title")
+
+    if name_param is None:
+        tasks = Task.query.all()
+    else:
+        tasks = Task.query.filter_by(title=name_param)
+    
+    response = []
+
+    for task in tasks:
+        if task.completed_at is None:
+            task_dict = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False
+            }
+            response.append(task_dict)
+        else:
+            task_dict = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "completed at": task.completed_at,
+                "is_complete": True}
+            response.append(task_dict)
+    return jsonify(response), 200
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    chosen_task = get_one_task_or_abort(task_id)
+
+    # if chosen_task.completed_at is None:
+    #     task = {
+    #         "id": chosen_task.task_id,
+    #         "title": chosen_task.title,
+    #         "description": chosen_task.description,
+    #         "is_complete": False
+    #     }
+
+    # else:
+    #     task = {
+    #         "id": chosen_task.task_id,
+    #         "title": chosen_task.title,
+    #         "description": chosen_task.description,
+    #         "completed at": chosen_task.completed_at,
+    #         "is_complete": True}
+        
+    return jsonify({"task": chosen_task.to_dict()}), 200
+
+
+
+
+#-------------------------------------POST------------------------------------------------
+@task_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or \
+        "description" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
Good that you're doing data validation here, but I suggest telling the end user what is invalid about the data.